lib/policies/basic-auth/auth.js

```diff
@@ -40,6 +40,9 @@ function authenticateBasic (req, clientId, clientSecret, done) {
     .catch(err => done(err));
 }
 
-module.exports = function () {
-  return passport.authenticate('basic', { session: false });
+module.exports = function (actionParams) {
+  return function (req, res, next) {
+    actionParams.session = false;
+    passport.authenticate('basic', actionParams, actionParams.getCommonAuthCallback(req, res, next))(req, res, next);
```

Comment:
third param allows custom handling of result.
By default calling done(null, false) will make auth to fail with 401. 
the passthrough allows to bypass that. 

alternative implementation is to use [basic, pass] strategies 
`pass` or `anonymous` strategies allows bypassing previous strategies (it calls `this.pass()`) 

but since 401 vs 403 handling is also required I've chosen the first variant